src/Carousel.tsx

```diff
@@ -67,10 +68,14 @@ class Carousel extends React.Component<CarouselProps, CarouselInternalState> {
     super(props);
     this.containerRef = React.createRef();
     this.listRef = React.createRef();
+    const childCount = React.Children.count(props.children);
     this.state = {
       itemWidth: 0,
       slidesToShow: 0,
-      currentSlide: 0,
+      currentSlide:
+        this.props.initialSlide < childCount
+          ? this.props.initialSlide
+          : childCount,
```

Comment:
```suggestion
          : 0,
```

Reply:
Does it make a difference if we just do 
```
currentSlide: props.initialSlide
```
That means currentSlide is 0 by default.